action/protocol/account/transfer.go

```diff
@@ -74,13 +77,21 @@ func (p *Protocol) handleTransfer(ctx context.Context, act action.Action, sm pro
 	if err != nil {
 		return nil, errors.Wrapf(err, "failed to decode recipient address %s", tsf.Recipient())
 	}
-	recipientAcct, err := accountutil.LoadAccount(sm, recipientAddr)
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to load address %s", tsf.Recipient())
+	recipientAcct, err := accountutil.LoadAccount(sm, recipientAddr, accountCreationOpts...)
+	if !fCtx.TolerateLegacyAddress {
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to load address %s", tsf.Recipient())
+		}
 	}
-	if recipientAcct.IsContract() {
-		// update sender Nonce
-		accountutil.SetNonce(tsf, sender)
+	fixNonce := protocol.MustGetFeatureCtx(ctx).FixGasAndNonceUpdate
+	blkCtx := protocol.MustGetBlockCtx(ctx)
```

Comment:
the name `FixGasAndNonceUpdate` is confusing

Reply:
`fixNonce` is initialized with feature `FixGasAndNonceUpdate`, which fixes both `nonce` and `gas`. Here, we only use the `fixNonce` attribute. Other than that, `fixNonce` is just a local parameter. I don't get it how could it be `confusing`.

Reply:
does it belong to pr `Skip update nonce and gas for system actions` ?

Reply:
we only have one PR for now.